src/test/java/com/idris/CustomerControllerTest.java

```diff
@@ -94,24 +76,93 @@ void callingGetCustomerShouldReturnAllFields() {
+
+        given()
+                .port(APP_PORT_NUMBER)
+                .when()
+                .delete("/api/v1/customers/1")
+                .then()
+                .statusCode(200)
+                .contentType(ContentType.JSON)
+                .body("message", is("Customer with ID 1 was deleted successfully."));
+    }
+    @Test
+    void shouldAddCustomerSuccessfully() {
+        String jsonString = """
+            {
+                "name": "Abdi Ali",
+                "email": "AbdiAli@example.com",
+                "age": 30
+            }
+            """;
+
+        given()
+                .port(APP_PORT_NUMBER)
+                .contentType(ContentType.JSON)
+                .body(jsonString)
+                .when()
+                .post("/api/v1/customers")
+                .then()
+                .statusCode(201)
+                .body("name", is("Abdi Ali"))
+                .body("email", is("AbdiAli@example.com"))
+                .body("age", is(30));
+    }
+
+    @Test
+    void shouldUpdateExistingCustomerFromJamesToAlice() throws JSONException {
         Jdbi jdbi = Jdbi.create(postgres.getJdbcUrl(), DATABASE_USER, DATABASE_PASSWORD);
-        jdbi.withHandle(handle -> handle.execute("INSERT INTO \"customer\" (id, \"name\", \"email\") VALUES (?, ?, ?)", 0, "Alice", "Alicce@gmail.com"));
+        jdbi.withHandle(handle -> handle.execute("INSERT INTO \"customer\" (id, \"name\", \"email\", age) VALUES (?, ?, ?, ?)", 2, "james", "james@gmail.com", 14));
+
+        String jsonString = """
+                {
+                    "id": 2,
+                    "name": "alice",
+                    "age": 20,
+                    "email": "alice@gmail.com"
+                }
+                """;
 
         given()
                 .port(APP_PORT_NUMBER)
+                .contentType(ContentType.JSON)
+                .body(jsonString)
                 .when()
-                .get("api/v1/customers")
+                .put("/api/v1/customers/2")
                 .then()
                 .statusCode(200)
-                .body("id", is(null))
-                .body("name", is(null))
-                .body("email", is(null));
+                .body("name", is("alice"))
+                .body("email", is("alice@gmail.com"))
+                .body("age", is(20));
+    }
+
+    @Test
+    void shouldReturn404WhenUpdatingNonExistentCustomer() {
+        int nonExistentCustomerId = 999;
 
+        String jsonString = """
+                {
+                    "name": "alice",
+                    "age": 20,
+                    "email": "alice@gmail.com"
+                }
+                """;
+
+        given()
+                .port(APP_PORT_NUMBER)
+                .contentType("application/json")
+                .body(jsonString)
+                .put("/api/v1/customers/" + nonExistentCustomerId)
```

Comment:
`"/api/v1/customers"` can be extracted out into a final static variable as you're using it across all your tests.